lib/lrama/grammar/parameterizing_rule_builder.rb

```diff
@@ -0,0 +1,47 @@
+module Lrama
+  class Grammar
+    class ParameterizingRuleBuilder
+      attr_reader :name, :args, :rhs, :term
+
+      def initialize(name, args, rhs)
+        @name = name
+        @args = args
+        @rhs = rhs
+        @term = nil
+        @required_args_count = args.count
+      end
+
+      def build_rules(token, build_token, rule_counter, lhs_tag, user_code, precedence_sym, line)
+        validate_argument_number!(token)
+        rules = []
+        @rhs.each do |rhs|
+          @term = rhs_term(token, rhs)
+          rules << Rule.new(id: rule_counter.increment, _lhs: build_token, _rhs: [@term].compact, lhs_tag: lhs_tag, token_code: rhs.user_code, precedence_sym: rhs.precedence_sym, lineno: line)
+        end
+        rules
+      end
+
+      def build_token(token)
+        validate_argument_number!(token)
+        Lrama::Lexer::Token::Ident.new(s_value: "#{name}_#{token.args.first&.s_value}")
```

Comment:
Generated rule name should include information about all arguments otherwise rule name conflict happens when
1. ParameterizingRule requires 2 or more arguments
2. Same ParameterizingRule is called with different arguments but their first argument is same.

For example,

```
%rule my_rule(X, Y) ...

%%

program: my_rule(number, number)
              | my_rule(number, string)
             ;
```

Reply:
I had missed that. Thank you! I have fixed it in the following commit:
https://github.com/ruby/lrama/pull/285/commits/74f43ae67195b2befddd933ace329a1648499c51